bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+        if (!(command instanceof RefreshType)) {
+            if (!config.key.isEmpty()) {
+                String param;
+                String channelID = channelUID.getId();
+                switch (channelID) {
+                    case CHANNEL_CONTROLBOILERAPPROVAL:
+                        param = getThing().getProperties().get(PARAMETER_BOILERAPPROVAL);
+                        break;
+                    case CHANNEL_CONTROLPROGRAM:
+                        param = getThing().getProperties().get(PARAMETER_PROGRAM);
+                        break;
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM0:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM1:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM2:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM3:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM4:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM5:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM6:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM7:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM8:
+                        param = getThing().getProperties().get(PARAMETER_HEATCIRCPROGRAM).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLWWHEAT0:
+                    case CHANNEL_CONTROLWWHEAT1:
+                    case CHANNEL_CONTROLWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_WWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLEXTRAWWHEAT0:
+                    case CHANNEL_CONTROLEXTRAWWHEAT1:
+                    case CHANNEL_CONTROLEXTRAWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_EXTRAWWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    default:
+                        return;
+                }
+                String response = sendGetRequest(PARSET_URL, "syn=" + param + "&value=" + command.toString());
+                // logger.warn("{} syn={}&value={}", PARSET_URL, param, command.toString());
+                State newState = new StringType(response);
+                updateState(channelID, newState);
+            } else {
+                logger.warn("A 'key' needs to be configured in order to control the Guntamatic Heating System");
+            }
+        }
+    }
+
+    private void parseAndUpdate(String html) {
+        String[] daqdata = html.split("\\n");
+
+        for (Integer i : channels.keySet()) {
+            String channel = channels.get(i);
+            String unit = units.get(i);
+            if ((channel != null) && (unit != null)) {
+                String value = daqdata[i].trim();
```

Comment:
you have first to check that daqdata.length > i

Reply:
`if ( (channel != null) && (unit != null) && (i < daqdata.lenght) ) {`